Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -2895,7 +2901,378 @@ namespace OpenInfraPlatform
+					{
+						double t = integrateParameter(polynomialConstantX, polynomialConstantY, parameter);
+						x = calculatePolynomialCurve(polynomialConstantX, t);
+						y = calculatePolynomialCurve(polynomialConstantY, t);
+						return carve::geom::VECTOR(x, y, 0.);
+					}
+					
+					else if (polynomialConstantCntX > 0 && polynomialConstantCntZ > 0)
+					{
+						double t = integrateParameter(polynomialConstantX, polynomialConstantZ, parameter);
+						x = calculatePolynomialCurve(polynomialConstantX, t);
+						z = calculatePolynomialCurve(polynomialConstantZ, t);
+						return carve::geom::VECTOR(x, 0., z);
+					}
+				
+					else if (polynomialConstantCntY > 0 && polynomialConstantCntZ > 0)
+					{
+						double t = integrateParameter(polynomialConstantY, polynomialConstantZ, parameter);
+						y = calculatePolynomialCurve(polynomialConstantY, t);
+						z = calculatePolynomialCurve(polynomialConstantZ, t);
+						return carve::geom::VECTOR(0., y, z);
+					}
+
+					else if (polynomialConstantCntX > 0 && polynomialConstantCntY > 0 && polynomialConstantCntZ > 0)
+					{
+						double t = integrateParameter(polynomialConstantX, polynomialConstantY, polynomialConstantZ, parameter);
+						x = calculatePolynomialCurve(polynomialConstantX, t);
+						y = calculatePolynomialCurve(polynomialConstantY, t);
+						z = calculatePolynomialCurve(polynomialConstantZ, t);
+						return carve::geom::VECTOR(x, y, z);
+					}
+				}
+
+				/*! \brief Calculates a polinomial curve
+				* \param[in] coefficient			A vector with coeffitients corresponding ordinate from \c IfcPolynomialCurve.
+				* \param[in] parameter				A length during the curve from \c IfcCurveSegment.
+				* \return							The location of the ordinate point.
+				* \note
+				*/
+				typename IfcEntityTypesT::IfcReal calculatePolynomialCurve(std::vector<double>& coefficient, const double& parameter) const noexcept(false)
+				{
+					int n = std::size(coefficient);
+					typename IfcEntityTypesT::IfcReal sum = coefficient[0];
+					double factor = 1;
+
+					for (int i = 1; i < n; i++)
+					{
+						factor *= parameter;
+						sum += coefficient[i] * factor;
+					}
+					return sum;
+				}
+
+				/*! \brief Calculates derivative of the polynomial curve
+				* \param[in] polynomialConstant	    A vector with coeffitients corresponding ordinate from \c IfcPolynomialCurve.
+				* \param[in] b				        Parameter value from intagrateParameter
+				* \return							Derivative value.
+				* \note
+				*/
+				double calculatePolynomialDerivative(std::vector<double>& polynomialConstant, double b)  const noexcept(false)
+				{
+					int polynomialConstantCnt = std::size(polynomialConstant);
+					double value = polynomialConstant[1];
+					int factor = 2;
+					// derivative of the polynomial
+					for (int i = 2; i < polynomialConstantCnt; i++)
+					{
+						value += polynomialConstant[i] * factor * b;
+						b *= b;
+						factor += 1;
+					}

```

Comment:
There might be a mistake here but I'm not sure
Lets say we want to have the derivative of the function 2 * b which would be 2 
It seems that this is not taken into account in this for loop because everything is multiplied with b 
Is this "first step" skipped on purpose?

Reply:
Thank you for you comment! 
In the code is only one mistake(!!), it does not work with vectors with size 1.
In this case we have constant polynomial with derivative equal 0.

For example:
*Example vector:*
**p3** = (1,2,3,4)
**polynomial** = 1 + 2*b + 3*b*b + 4*b*b*b
**derivative by hand** = 2 + 6*b + 12*b*b

I skiped first step because the first member of the polynomialConstant vector is linear term which derivative is eqaul to polynomialConstant[1]
And thats why I start with the second iteration .

*With code:*
value = 2;
faktor = 2;
`i = 2`
**value** = 2 + 3*2*b 
b = b*b
faktor = 3
`i = 3`
**value** = 2 + 3*2*b + 4*3*b*b
b = b*b*b
faktor = 4

Resolved in [102578f](https://github.com/tumcms/Open-Infra-Platform/pull/539/commits/102578f6c6f89ea5cb99c511be0ab4b573f31c0a)